main.py

```diff
@@ -0,0 +1,72 @@
+from flask import Flask, request, jsonify, abort
+from flask_sqlalchemy import SQLAlchemy
+from flask_marshmallow import Marshmallow
+
+app = Flask(__name__)
+app.config['SQLALCHEMY_DATABASE_URI'] = "mysql+mysqlconnector://iotuser:iotuser@localhost/iot-test-db"
+app.config['SQLALCHEMY_TRACK_MODIFICATIONS'] = False
+db = SQLAlchemy(app)
+ma = Marshmallow(app)
+
+
+class Recipe(db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    name = db.Column(db.String(64), unique=False)
+    number_of_dishes = db.Column(db.Integer, unique=False)
+    caloric = db.Column(db.Integer, unique=False)
+
+
+class RecipeSchema(ma.Schema):
+    class Meta:
+        fields = ('id', 'name', 'number_of_dishes', 'caloric')
+
+
+@app.route("/", methods=["GET"])
+def get_recipe():
+    recipes = Recipe.query.all()
+    return jsonify(recipes_schema.dump(recipes))
+
+
+@app.route("/<id>", methods=["GET"])
+def get_recipe(id):
+    recipe = Recipe.query.get(id)
+    if not recipe:
+        abort(404)
+    return jsonify(recipe_schema.dump(recipe))
+
+
+@app.route("/", methods=["POST"])
+def add_recipe():
+    recipe = Recipe(name=request.json["name"], number_of_dishes=request.json["number_of_dishes"], caloric=request.json["caloric"])
```

Comment:
This is going to throw HTTP 500s on invalid user input (e.g. missing "name"). Please validate the user input using Marshmallow, and return HTTP400 for invalid input.